nx/lib/nx.ex

```diff
@@ -7046,6 +7046,123 @@ defmodule Nx do
     impl!(tensor).sort(tensor, tensor, axis: axis, comparator: comparator)
   end
 
+  @doc """
+  Calculates the QR decomposition of an 2-D tensor with shape `{M, N}` via the Householder method.
+
+  ## Options
+
+  `:mode` - Can be one of `:reduced`, `:complete`, `:r` or `:raw`. Defaults to `:reduced`
+    For the following, `K = min(M, N)`
+
+    * `:reduced` - returns `q` and `r` with shapes `{M, K}` and `{K, N}`
+    * `:complete` - returns `q` and `r` with shapes `{M, M}` and `{M, N}`
+
+  `:q_axes` - Defines the axes for the `q` tensor
```

Comment:
We use `names` elsewhere so I would just call these `q_names` and `r_names` respectively